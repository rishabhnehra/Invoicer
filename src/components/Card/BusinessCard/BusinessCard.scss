.BusinessCard {
    border-radius: 10px;
    padding: 16px;
    color: #FFF;
    width: 320px;
    background-color: #9E579D;
    box-shadow: 0px 0px 6px 3px rgba(0,0,0,0.1);

    h5 {
        color:#D1D1D1;
        font-weight: 400;
        margin-bottom: 4px;
    }

    h1 {
        font-size: 18px;
        font-weight: 700;
        margin-bottom: 2px;
    }

    h2 {
        font-size: 14px;
        font-weight: 600;
        margin-bottom: 2px;
    }

    p {
        font-size: 13px;
        font-weight: 100;
    }
}