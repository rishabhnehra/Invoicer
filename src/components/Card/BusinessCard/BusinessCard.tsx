export type BusinessCardProps = {
    isSender: boolean,
    name: string,
    gst: string,
    address: string,
    phone: number
};

export const BusinessCard : React.FC<BusinessCardProps> = ({
    isSender = false,
    name = 'NA',
    gst = 'NA',
    address = 'NA',
    phone = 123
}) => <div className="BusinessCard">
    <h5>{isSender ? 'Sender' : 'Receiver'}</h5>
    <h1>{name}</h1>
    <h2>{gst}</h2>
    <p>{address}</p>
    <p>Phone: {phone}</p>
</div>