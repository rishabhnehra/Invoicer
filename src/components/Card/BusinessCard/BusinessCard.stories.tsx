import { Story, Meta } from '@storybook/react/types-6-0';

import {BusinessCardProps, BusinessCard as BusinessCardComponent} from './BusinessCard';

export default {
    title: 'Components/Card',
    component: BusinessCardComponent,
} as Meta;

const Template: Story<BusinessCardProps> = (args) => <BusinessCardComponent {...args} />;

export const BusinessCard = Template.bind({});
BusinessCard.args = {
    isSender: true,
    name: 'Rishabh Controls',
    gst: '24ABNPN1124A1ZA',
    address: `Address #1
    Address #2`,
    phone: 1234567890
}