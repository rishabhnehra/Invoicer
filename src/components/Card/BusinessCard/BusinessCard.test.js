import renderer from 'react-test-renderer';
import { BusinessCard } from "./BusinessCard";

describe("<BusinessCard />", () => {
    it("snapshot check", () => {
        const tree = renderer.create(<BusinessCard 
            isSender={true}
            name='Rishabh Controls'
            gst= '24ABNPN1124A1ZA'
            address= {`Address #1
            Address #2`}
            phone={1234567890}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});