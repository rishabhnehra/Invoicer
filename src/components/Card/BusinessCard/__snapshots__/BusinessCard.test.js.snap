// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessCard /> snapshot check 1`] = `
<div
  className="BusinessCard"
>
  <h5>
    Sender
  </h5>
  <h1>
    Rishabh Controls
  </h1>
  <h2>
    24ABNPN1124A1ZA
  </h2>
  <p>
    Address #1
            Address #2
  </p>
  <p>
    Phone: 
    1234567890
  </p>
</div>
`;
